dvc/repo/gc.py

```diff
@@ -74,40 +76,46 @@ def gc(
     if repos:
         all_repos.extend(Repo(path) for path in repos)
 
-    all_clists = _load_all_used_cache(
-        all_repos,
-        target=None,
-        all_branches=all_branches,
-        active=False,
-        with_deps=with_deps,
-        all_tags=all_tags,
-        remote=remote,
-        force=force,
-        jobs=jobs,
-    )
-
-    if len(all_clists) > 1:
-        clist = _merge_cache_lists(all_clists)
-    else:
-        clist = all_clists[0]
-
-    with self.state:
-        _do_gc("local", self.cache.local.gc, clist)
-
-        if self.cache.s3:
-            _do_gc("s3", self.cache.s3.gc, clist)
-
-        if self.cache.gs:
-            _do_gc("gs", self.cache.gs.gc, clist)
-
-        if self.cache.ssh:
-            _do_gc("ssh", self.cache.ssh.gc, clist)
-
-        if self.cache.hdfs:
-            _do_gc("hdfs", self.cache.hdfs.gc, clist)
-
-        if self.cache.azure:
-            _do_gc("azure", self.cache.azure.gc, clist)
-
-        if cloud:
-            _do_gc("remote", self.cloud.get_remote(remote, "gc -c").gc, clist)
+    with ExitStack() as stack:
```

Comment:
what is happening here? could you please explain why do we need ExitStack? may be put a comment as well